software/src/master/src/kernel/V_VThread.cpp

```diff
@@ -195,3 +197,41 @@ bool V::VThread::stop () {
     return pthread_cancel (m_hThread) == 0;
 #endif
 }
+
+int const V::VThread::m_VSingleCoreExecution = V::GetEnvironmentInteger ("VSingleCoreExecution", 0);
+int       V::VThread::m_VSingleCoreExecutionChecked = 0;
+
+void V::VThread::CheckSingleCoreExecution () {
+    if (m_VSingleCoreExecutionChecked == 0) {
+        if (m_VSingleCoreExecution != 0)
+            SetSingleCoreExecution();
+    }
+
+    m_VSingleCoreExecutionChecked = 1;
+}
+
+int V::VThread::SetSingleCoreExecution () {
+    bool success = -1;
+
+#if defined(V_THREAD_ENABLED) && defined (__linux__)
+
+    // Determine how many CPUs exist in order to randomly assign appropriately
+    const int cpu_count = sysconf(_SC_NPROCESSORS_ONLN);
+    
+    // get our random-ish CPU index
```

Comment:
[V::VRandom::BoundedValue<T>](https://github.com/vision-dbms/vision/blob/f37400b0161e97e8de07c8a9fa9dea266c734bf9/software/src/master/src/kernel/V_VRandom.h#L56) implements this logic